packages/composite-checkout-wpcom/package.json

```diff
@@ -36,14 +36,14 @@
 	},
 	"homepage": "https://github.com/Automattic/wp-calypso/tree/master/packages/composite-checkout-wpcom#readme",
 	"dependencies": {
+		"@automattic/calypso-polyfills": "file:../calypso-polyfills",
```

Comment:
This is controversial. `calypso-polyfills` is not used by the package code itself, but only by a mock module useful only for tests (`mock/cart-endpoint.ts`).

An app that uses the `composite-checkout-wpcom` package certainly never uses the mock and the `calypso-polyfills` dependency is never used.

The `calypso-polyfills` package makes sense only for apps, and it works correctly only if it's the first thing ever imported by app's main file. A package should never depend on it.

@sirbrillig @nbloomf what if the `mock/cart-endpoint.ts` module was moved to Calypso, to a place near the only usage, the `client/my-sites/checkout/checkout/test/composite-checkout.js` file? Or do you think it's useful also for other consumers?

Reply:
Yeah, it doesn't make sense to have a `dependency` on something that is only used for tests. That should be a `devDependency`, so maybe I should move this to the root?

Reply:
> useful also for other consumers

Unlike `composite-checkout`, the `composite-checkout-wpcom` package effectively is just a bunch of calypso things. It was only put in a separate package for the purposes of modularization, but it is extremely hard-tied to calypso. I've been planning for months to move some or the whole thing into calypso proper as the separation has been making a lot of things challenging; maybe this is a good motivation to increase the priority of that refactor.

In the short term, I'd say if you need to drop a dependency in this package, that's fine; just disable any tests it breaks and let us know.

Reply:
For reference, this dependency is being removed in #40596.